Bugsnag/Helpers/BSGRunContext.m

```diff
@@ -272,12 +281,22 @@ static void AddObservers() {
 }
 
 
-#pragma mark -
+#pragma mark - Misc
 
 void BSGRunContextUpdateTimestamp() {
     bsg_runContext->timestamp = CFAbsoluteTimeGetCurrent();
 }
 
+static void UpdateAvailableMemory() {
+    // Deliberately avoids use of bsg_ksmachfreeMemory() because that falls back
+    // to a much more expensive (~5x) system call on earlier releases.
```

Comment:
Just to be clear, what I understand from the comments and `bsg_ksmachfreeMemory` code, is that freeMemory will be reported from the quicker `os_proc_available_memory` API if available, but on simulators or older devices the crash report falls back to a slower API but the OOM report will omit the field.

I see the check in `BSG_KSMach.c` does a check for a 0 response to see if "it's working" - would we want to do this here?

Reply:
On simulators or older devices OOMs will not include `freeMemory`.

`bsg_ksmachfreeMemory()` falls back to using the `free_count` from the `HOST_VM_INFO` flavor of `host_statistics()`, which returns system-wide VM statistics, whereas `os_proc_available_memory()` returns the amount remaining in the current process' dirty memory limit.

The two values could be wildly different given that memory limits are typically 50-80% of system memory for foreground apps, and much smaller for app extensions. Using the fallback could be confusing or misleading in addition to being much slower, since it's hitting the memory limit that causes apps to be killed.

The check for `0` is done when constructing the event - [here](https://github.com/bugsnag/bugsnag-cocoa/pull/1357/files#diff-9e40e69405013ebceec8e93a7bb9f77586d4fbecc564622b874b364e0b772d29R1129).